tunslip6.c

```diff
@@ -363,6 +375,119 @@ serial_to_tun(FILE *inslip, int outfd)
 unsigned char slip_buf[2000];
 int slip_end, slip_begin;
 
+int
+get_slipfd()
+{
+  const int start_i = 20;
+  int status, i, fd = -1;
+  struct timeval sleep_for = {
+    .tv_sec = 0,       /* seconds */
+    .tv_usec = 200000  /* microseconds */
+  };
+
+  /* Try to acquire slipfd a few times */
+  for (i = start_i; i > 0; --i) {
+    if (i != start_i) {
+      select(0, NULL, NULL, NULL, &sleep_for);
+    }
+
+    if(host != NULL) {
+      struct addrinfo hints, *servinfo, *p;
+      int rv;
+      char s[INET6_ADDRSTRLEN];
+
+      if(port == NULL) {
+        port = "60001";
+      }
+
+      memset(&hints, 0, sizeof hints);
+      hints.ai_family = AF_UNSPEC;
+      hints.ai_socktype = SOCK_STREAM;
+
+      if((rv = getaddrinfo(host, port, &hints, &servinfo)) != 0) {
+        fprintf(stderr, "getaddrinfo: %s", gai_strerror(rv));
+        continue;
+      }
+
+      /* loop through all the results and connect to the first we can */
+      for(p = servinfo; p != NULL; p = p->ai_next) {
+        if((fd = socket(p->ai_family, p->ai_socktype,
+                            p->ai_protocol)) == -1) {
+          perror("client: socket");
+          continue;
+        }
+
+        if(connect(fd, p->ai_addr, p->ai_addrlen) == -1) {
+          close(fd);
+          perror("client: connect");
+          continue;
+        }
+        break;
+      }
+
+      if(p == NULL) {
+        fprintf(stderr, "can't connect to ``%s:%s''\n", host, port);
+        continue;
+      }
+
+      fcntl(fd, F_SETFL, O_NONBLOCK);
+
+      inet_ntop(p->ai_family, get_in_addr((struct sockaddr *)p->ai_addr),
+                s, sizeof(s));
+      fprintf(stderr, "slip connected to ``%s:%s''\n", s, port);
+
+      /* all done with this structure */
+      freeaddrinfo(servinfo);
+
+    } else {
+      if(siodev != NULL) {
+        fd = devopen(siodev, O_RDWR | O_NONBLOCK);
+        if(fd == -1) {
+          fprintf(stderr, "can't open siodev ``%s''\n", siodev);
+          continue;
+        }
+      } else {
+        static const char *siodevs[] = {
+          "ttyUSB0", "cuaU0", "ucom0" /* linux, fbsd6, fbsd5 */
+        };
+        int i;
+        for(i = 0; i < 3; i++) {
+          siodev = siodevs[i];
+          fd = devopen(siodev, O_RDWR | O_NONBLOCK);
+          if(fd != -1) {
+            break;
+          }
+        }
+        if(fd == -1) {
+          fprintf(stderr, "can't open siodev\n");
+          continue;
+        }
+      }
+      if (timestamp) stamptime();
+      if (siodev[0] != '/') {
```

Comment:
@jukkar I reused the logic from devopen() so this should produce correct output now (working for me)